libwget/io.c

```diff
@@ -433,12 +433,13 @@ int wget_update_file(const char *fname,
 		if (fp) {
 			// read fname data
 			if (load_func(context, fp)) {
-				fclose(fp);
+				if (! fclose(fp))
+					fp = NULL;
```

Comment:
`fp` is a local variable and we return two lines below here. No need to set `fp` to `NULL`.